### DISCUSSION TITLE
Preload relations in BaseModel and global where

### DISCUSSION THREAD

**Discussion by vitortoigo** [adonisjs/core#4255] [Help]
Hello,
I'm doing a tenancy based on a shared database, but i has some questions:

**1 -** Have a way to assign a `where` in the builder instace every time when a fetch is used (in any model)? Like in a middleware?
**2 -** Has a easy way to preload relations in the model? (In laravel we have a public variable $with, just assign the relation names and in every query with the model, returns the relation);

My working code:
````
export default class DefaultModel extends BaseModel {
    @column()
    tenant_id: number;

    @belongsTo(() => Tenant, {
        foreignKey: 'tenant_id',
    })
    public tenant: BelongsTo<typeof Tenant>;

    @beforeCreate()
    public static setTenant(model) {
        const ctx = HttpContext.get()!;
        model.tenant_id = ctx.auth.user?.tenant.id;
    }

    @beforeFetch()
    public static getTenant(query) {
        const ctx = HttpContext.get()!;
        query.where('tenant_id', ctx.auth.user?.tenant.id);
    }
}
````

Basically, i extends this custom Model to anothers to use same methods.
